does not work on pypy3 (although the fix should be easy)

I reported this in the pypy compatibility tracker here:

https://bitbucket.org/pypy/compatibility/issues/8/psycopg2-does-not-work-on-pypy3

For your convenience, the errors are here:

```
    cc -pthread -arch x86_64 -DNDEBUG -O2 -L/usr/local/Cellar/openssl@1.1/1.1.0g/lib -L/usr/local/opt/openssl@1.1/lib -L/lib -I/usr/local/Cellar/openssl@1.1/1.1.0g/include -I/usr/local/opt/openssl@1.1/include -I/include -L/lib -I/include -I/usr/local/Cellar/openssl@1.1/1.1.0g/include -I/usr/local/opt/openssl@1.1/include -I/include -fPIC -DPSYCOPG_DEFAULT_PYDATETIME=1 -DPSYCOPG_VERSION="2.7.3.2 (dt dec pq3 ext lo64)" -DPG_VERSION_NUM=100000 -DHAVE_LO64=1 -I/usr/local/Cellar/pypy3/5.9.0/libexec/include -I. -I/usr/local/Cellar/postgresql/10.0/include -I/usr/local/Cellar/postgresql/10.0/include/server -c psycopg/adapter_datetime.c -o build/temp.macosx-10.13-x86_64-3.5/psycopg/adapter_datetime.o
    clang: warning: argument unused during compilation: '-L/usr/local/Cellar/openssl@1.1/1.1.0g/lib' [-Wunused-command-line-argument]
    clang: warning: argument unused during compilation: '-L/usr/local/opt/openssl@1.1/lib' [-Wunused-command-line-argument]
    clang: warning: argument unused during compilation: '-L/lib' [-Wunused-command-line-argument]
    clang: warning: argument unused during compilation: '-L/lib' [-Wunused-command-line-argument]
    psycopg/adapter_datetime.c:103:18: error: no member named 'microseconds' in 'PyDateTime_Delta'
        int a = obj->microseconds;
                ~~~  ^
    psycopg/adapter_datetime.c:112:34: error: no member named 'days' in 'PyDateTime_Delta'
                                obj->days, obj->seconds, buffer);
                                ~~~  ^
    psycopg/adapter_datetime.c:112:45: error: no member named 'seconds' in 'PyDateTime_Delta'
                                obj->days, obj->seconds, buffer);
                                           ~~~  ^
    3 errors generated.
    error: command 'cc' failed with exit status 1
```

The pypy folks tell me that the issue here is that the fields in `PyDateTime` are private.  As documented in the CPython official docs, the way to get at these values is via macro accessors; for example: https://docs.python.org/3.6/c-api/datetime.html#c.PyDateTime_DATE_GET_MICROSECOND, which PyPy3 does implement.

--- a/psycopg/adapter_datetime.c
+++ b/psycopg/adapter_datetime.c
@@ -100,7 +100,7 @@ _pydatetime_string_delta(pydatetimeObject *self)
 
     char buffer[8];
     int i;
-    int a = obj->microseconds;
+    int a = PyDateTime_DELTA_GET_MICROSECONDS(obj);
 
     for (i=0; i < 6 ; i++) {
         buffer[5-i] = '0' + (a % 10);
@@ -109,7 +109,9 @@ _pydatetime_string_delta(pydatetimeObject *self)
     buffer[6] = '\0';
 
     return Bytes_FromFormat("'%d days %d.%s seconds'::interval",
-                            obj->days, obj->seconds, buffer);
+                            PyDateTime_DELTA_GET_DAYS(obj),
+                            PyDateTime_DELTA_GET_SECONDS(obj),
+                            buffer);
 }
 
 static PyObject *

--- a/psycopg/adapter_datetime.c
+++ b/psycopg/adapter_datetime.c
@@ -93,12 +93,6 @@ error:
     return rv;
 }
 
-#ifndef PyDateTime_DELTA_GET_DAYS
-#define PyDateTime_DELTA_GET_DAYS(o) (o->days)
-#define PyDateTime_DELTA_GET_SECONDS(o) (o->seconds)
-#define PyDateTime_DELTA_GET_MICROSECONDS(o) (o->microseconds)
-#endif
-
 static PyObject *
 _pydatetime_string_delta(pydatetimeObject *self)
 {

--- a/psycopg/adapter_datetime.c
+++ b/psycopg/adapter_datetime.c
@@ -93,6 +93,12 @@ error:
     return rv;
 }
 
+#ifndef PyDateTime_DELTA_GET_DAYS
+#define PyDateTime_DELTA_GET_DAYS(o) (o->days)
+#define PyDateTime_DELTA_GET_SECONDS(o) (o->seconds)
+#define PyDateTime_DELTA_GET_MICROSECONDS(o) (o->microseconds)
+#endif
+
 static PyObject *
 _pydatetime_string_delta(pydatetimeObject *self)
 {

--- a/psycopg/python.h
+++ b/psycopg/python.h
@@ -87,6 +87,7 @@ typedef unsigned long Py_uhash_t;
 #ifndef PyNumber_Int
 #define PyNumber_Int           PyNumber_Long
 #endif
+
 #endif  /* PY_MAJOR_VERSION > 2 */
 
 #if PY_MAJOR_VERSION < 3
@@ -103,6 +104,10 @@ typedef unsigned long Py_uhash_t;
 #define Bytes_FromFormat PyString_FromFormat
 #define Bytes_ConcatAndDel PyString_ConcatAndDel
 #define _Bytes_Resize _PyString_Resize
+
+#define PyDateTime_DELTA_GET_DAYS(o)         (((PyDateTime_Delta*)o)->days)
+#define PyDateTime_DELTA_GET_SECONDS(o)      (((PyDateTime_Delta*)o)->seconds)
+#define PyDateTime_DELTA_GET_MICROSECONDS(o) (((PyDateTime_Delta*)o)->microseconds)
 
 #else
 

